Directory.Build.props

```diff
@@ -80,14 +85,15 @@
     <ExcludePartialFacadesImport>true</ExcludePartialFacadesImport>
     <ExcludeApiCompatImport>true</ExcludeApiCompatImport>
     <ExcludeReferenceAssembliesImport>true</ExcludeReferenceAssembliesImport>
+    <ExcludeCodeAnalysisImport>true</ExcludeCodeAnalysisImport>
   </PropertyGroup>
 
   <!-- Import Build tools common props file where repo-independent properties are found -->
   <Import Project="$(ToolsDir)Build.Common.props" Condition="Exists('$(ToolsDir)Build.Common.props')" />
 
   <!-- Enable the analyzers for this repo -->
   <PropertyGroup>
-    <EnableDotnetAnalyzers Condition="'$(EnableDotnetAnalyzers)'==''">true</EnableDotnetAnalyzers>
+    <EnableAnalyzers Condition="'$(EnableAnalyzers)'=='' AND $([System.Text.RegularExpressions.Regex]::IsMatch($(MSBuildProjectDirectory), 'src%24'))">true</EnableAnalyzers>
```

Comment:
We should also define a property for `IsSourceProject` same as we have for `IsTestProject` and `IsReferenceAssembly`. We would also use that in https://github.com/dotnet/corefx/issues/30177#issuecomment-431120974

Reply:
Yeah, I will update, just copied from what it was from buildtools!